lib/SparkPost/APIResource.php

```diff
@@ -170,47 +172,38 @@ public static function deleteResource( $resourcePath=null, $options=array() ) {
 	 * @param array $options (optional) query string parameters
 	 * @return array Result set of action performed on resource
 	 */
-	private static function callResource( $action, $resourcePath=null, $options=array() ) {
-		
-		if( !in_array( $action, array('get', 'delete') ) ) throw new \Exception('Invalid resource action');
-		
-		//build the url
-		$hostConfig = SparkPost::getConfig();
-		$url = self::getBaseUrl($hostConfig);
-		if (!is_null($resourcePath)){
-			$url .= '/'.$resourcePath;
-		}
-		
-		// untested:
-		if( !empty($options) ) {
-			$queryString = http_build_query($options);
-			$url .= '?'.$queryString;
+	private function callResource( $action, $resourcePath=null, $options=[] ) {
+		$action = strtoupper($action); // normalize
+
+		if( !in_array($action, ['POST', 'PUT', 'GET', 'DELETE'])) {
+			throw new \Exception('Invalid resource action');
```

Comment:
This also isn't used, its kind of just defensive but this can never actually be called since call resource is private.